service-base/src/test/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecksTest.java

```diff
@@ -108,37 +102,43 @@ public void setup() {
      * @param ctx The vert.x test context.
      */
     @Test
-    public void testConnectionLimitIsNotReached(final TestContext ctx) {
+    public void testConnectionLimitIsNotReached(final VertxTestContext ctx) {
 
         givenCurrentConnections(9);
         final JsonObject limitsConfig = new JsonObject()
                 .put(PrometheusBasedResourceLimitChecks.FIELD_MAX_CONNECTIONS, 10);
         final TenantObject tenant = TenantObject.from(Constants.DEFAULT_TENANT, true);
         tenant.setProperty(TenantConstants.FIELD_RESOURCE_LIMITS, limitsConfig);
-        limitChecksImpl.isConnectionLimitReached(tenant).setHandler(ctx.asyncAssertSuccess(b -> {
-            ctx.assertEquals(Boolean.FALSE, b);
-            verify(webClient).get(eq(DEFAULT_PORT), eq(DEFAULT_HOST), anyString());
-        }));
+
+        limitChecksImpl.isConnectionLimitReached(tenant).setHandler(
+                ctx.succeeding(response -> ctx.verify(() -> {
+                    assertEquals(Boolean.FALSE, response);
+                    verify(webClient).get(eq(DEFAULT_PORT), eq(DEFAULT_HOST), anyString());
+                    ctx.completeNow();
```

Comment:
FMPOV the context should be completed outside of the verify block in general, because you might want to perform other checks as well that need to succeed. However, in this case there is no difference in the result ...